packages/itwin/grouping-mapping-widget/src/widget/components/context/PropertyGridWrapperContext.ts

```diff
@@ -4,29 +4,24 @@
 *--------------------------------------------------------------------------------------------*/
 import * as React from "react";
 import type { PropertyRecord } from "@itwin/appui-abstract";
-import { QueryBuilder } from "../QueryBuilder";
+import type { QueryBuilder } from "../QueryBuilder";
 
 export interface PropertySelection {
   currentPropertyList: PropertyRecord[];
   setCurrentPropertyList: React.Dispatch<
   React.SetStateAction<PropertyRecord[]>
   >;
-  query: string;
   setQuery: React.Dispatch<React.SetStateAction<string>>;
-  queryBuilder: QueryBuilder;
-  setQueryBuilder: React.Dispatch<React.SetStateAction<QueryBuilder>>;
-  isLoading: boolean;
-  isRendering: boolean;
+  queryBuilder?: QueryBuilder;
+  setQueryBuilder: React.Dispatch<React.SetStateAction<QueryBuilder | undefined>>;
```

Comment:
Why the type difference between `PropertySelection.setQueryBuilder` and `PropertyGridWrapperContext.setQueryBuilder`? These are the same functions. Simplify `PropertySelection.setQueryBuilder`.

Reply:
Same applies to `setQuery`.

Reply:
Same applies to `setCurrentPropertyList`.

Reply:
Changed